problem_builder/public/js/student_answers_dashboard.js

```diff
@@ -0,0 +1,299 @@
+                    this._enable(forward);
+                } else if (currentPage === totalPages) {
+                    this._enable(backward);
+                    this._disable(forward);
+                } else {
+                    this._enable(all);
+                }
+            }
+        },
+
+        _enable: function(controls) {
+            _.each(controls, function(control) {
+                this.$(control).prop('disabled', false);
+            }, this);
+        },
+
+        _disable: function(controls) {
+            _.each(controls, function(control) {
+                this.$(control).prop('disabled', true);
+            }, this);
+        }
+
+    });
+
+    var resultsView = new ResultsView({
+        collection: new Results([], { mode: "client", state: { pageSize: 15 } }),
+        el: $element.find('#results')
+    });
+
+    // Set up gettext in case it isn't available in the client runtime:
+    if (typeof gettext == "undefined") {
+        window.gettext = function gettext_stub(string) { return string; };
+        window.ngettext = function ngettext_stub(strA, strB, n) { return n == 1 ? strA : strB; };
+    }
+    var $startButton = $element.find('.data-export-start');
+    var $cancelButton = $element.find('.data-export-cancel');
+    var $downloadButton = $element.find('.data-export-download');
+    var $deleteButton = $element.find('.data-export-delete');
+    var $blockTypes = $element.find("select[name='block_types']");
+    var $rootBlockId = $element.find("input[name='root_block_id']");
+    var $username = $element.find("input[name='username']");
+    var $matchString = $element.find("input[name='match_string']");
+    var $resultTable = $element.find('.data-export-results');
+
+    var status;
+    function getStatus() {
+        $.ajax({
+            type: 'POST',
+            url: runtime.handlerUrl(element, 'get_status'),
+            data: '{}',
+            success: updateStatus,
+            dataType: 'json'
+        });
+    }
+
+    function updateStatus(newStatus) {
+        var statusChanged = newStatus !== status;
+        status = newStatus;
+        if (status.export_pending) {
+            // Keep polling for status updates when an export is running.
+            setTimeout(getStatus, 1000);
+        }
+        if (statusChanged) updateView();
+    }
+
+    function showSpinner() {
+        $startButton.prop('disabled', true);
+        $cancelButton.prop('disabled', true);
+        $downloadButton.prop('disabled', true);
+        $deleteButton.prop('disabled', true);
+        $('.data-export-status', $element).empty().append(
+            $('<i>').addClass('icon fa fa-spinner fa-spin')
+        ).css("text-align", "center");
+    }
+
+    function hideResults() {
+        $resultTable.hide();
+    }
+
+    function showResults() {
+        $resultTable.show();
+    }
+
+    function maybeShowResults() {
+        if (status.last_export_result) {
+            showResults();
+        }
+    }
```

Comment:
Nit: I think you can combine this with `showResults`, since there doesn't seem to be any time when you want to force the resultTable to appear before the last_export_result is ready.